### DISCUSSION TITLE
Week 7 Post

### DISCUSSION THREAD

**Discussion by Skye404** [USU-ITLS-x265-23-summer/class-repo#147] [General]
I realized while trying to complete the ES6 section this week that I was missing some tasks in the Basic JavaScript section, so I went back to work on those. Learning those basic things was really helpful and helped answer some lingering questions I've had. I thought I was caught up, so I'm not sure how I missed those tasks. 

The debugging section was really helpful to see some frequent errors to avoid, like a space or symbol that shouldn't be there or forgetting the ending parentheses. I think mostly what I want to do when creating my own project is remember to double check the code for anything that is out of place. I think I'm more subject to using the wrong tag or variable than forgetting a parentheses. I feel like I'm pretty good at paying attention to details, and my main issue is not knowing exactly which details to use to get the result I want.

Instructional Aspects: attention to detail, examples provided, obvious help tools on the page, perhaps a frequently asked questions page?

Project. @metasean I think I'd like most to try to make my own website/portfolio for writing and designs. This feels very ambitious. Am I going to be able to create this in a way that is actually professional looking? If we are creating our projects within GitHub, can I translate this code to use it outside of GitHub?

**Comment by derekscholes** [comment]
@Skye404 Something I have found with GitHub is that you can translate the code to use it outside of GitHub. One of the major things I like to do is use [Visual Studio Code](https://code.visualstudio.com) because it can connect directly to GitHub codespaces so you can edit there instead of on the web browser. In addition, it makes it very easy to commit and push requests from there. Something I also like is that you can copy and past and use a separate Visual Studio tab to create a live output from saved files on your computer. So what I tend to do is have one set of files on my computer, edit them and see the output change live, then I copy and paste it into my codespace and commit it. That way I have less errors in my codespace and can easily resolve any problems that occur. 

Here is an example of using my own files with the live output. 
<img width="1136" alt="Screenshot 2023-06-28 at 7 23 10 AM" src="https://github.com/USU-ITLS-x265-23-summer/class-repo/assets/132800749/31cf68cd-2e60-45e2-990b-dfda7e589709">
You just download the Go Live extension and can click on **Go Live** in the bottom on the screen. It should then bring up the live browser.
<img width="1680" alt="Screenshot 2023-06-28 at 7 23 42 AM" src="https://github.com/USU-ITLS-x265-23-summer/class-repo/assets/132800749/4a60525b-48ba-4ac5-a0b2-29d78dab0d47">

**Reply by Skye404** [reply]
Thanks, I'll have to look into this more next week.

**Comment by bst3ck** [comment]
I have never used the Go Live extension, I am going to have to try that.

Another option that you could use is to install a program on your computer called MAMP (for a Mac). It will allow you to create a virtual host on your computer and you can view your files like you would if they were already deployed to your hosting server. The nice thing about MAMP is that it allows some advanced features like MySQL and can run PHP backend services so you can mock up POST and GET requests which then can process them like a webserver. It maybe more advanced then what we need, but it has helped me in the past.
